EIPS/eip-1153.md

```diff
@@ -43,22 +55,36 @@ They use the same arguments on stack as `SLOAD` (`0x54`) and `SSTORE` (`0x55`).
 
 Addressing is the same as `SLOAD` and `SSTORE`. i.e. each 32-byte address points to a unique 32-byte word.
 
-Gas cost for both is 8 units of gas, regardless of values stored.
+Gas cost for `TSTORE` is the same as a hot dirty `SSTORE` (original value is not new value and is not current value, currently 100 gas), 
+and gas cost of `TLOAD` is the same as a hot `SLOAD` (value has been read before, currently 100 gas). Gas cost cannot be on par with memory access due to transient storage's interactions with reverts.
 
-The effects of transient storage are discarded at the end of the transaction.
+All values in transient storage are discarded at the end of the transaction.
 
-Transient storage is private to the contract that owns it, in the same way as "regular" storage is. Only owning contract frames may access their transient storage. And when they do, all the frames access the same transient store, in the same way as "regular" storage, but unlike "memory".
+Transient storage is private to the contract that owns it, in the same way as persistent storage. 
+Only owning contract frames may access their transient storage. 
+And when they do, all the frames access the same transient store, in the same way as persistent storage, but unlike memory.
 
-When transient storage is used in the context of `DELEGATECALL` or `CALLCODE`, then the owning contract of the transient storage is the contract that issued `DELEGATECALL` or `CALLCODE` instruction (the caller). When transient storage is used in the context of `CALL` or `STATICCALL`, then the owning contract of the transient storage is the contract that is the target of the `CALL` or `STATICCALL` instruction (the callee).
+When transient storage is used in the context of `DELEGATECALL` or `CALLCODE`, then the owning contract of the transient
+storage is the contract that issued `DELEGATECALL` or `CALLCODE` instruction (the caller) as with persistent storage. 
+When transient storage is used in the context of `CALL` or `STATICCALL`, then the owning contract of the transient storage
+is the contract that is the target of the `CALL` or `STATICCALL` instruction (the callee).
 
-Transient storage does not interact with reverts or invalid transactions, that means if a frame reverts, its effects on the transient storage remain until the end of the transaction.
+If a frame reverts, all writes to transient storage within that frame are also reverted, as with persistent storage.
 
 ## Rationale
-There is a proposal to alleviate the cost of inter-frame communication by reducing the cost of `SSTORE` when it modifies the same item multiple times within the same transaction (EIP-1087).
+There is a proposal to alleviate the cost of inter-frame communication by reducing the cost of `SSTORE` 
+when it modifies the same item multiple times within the same transaction ([EIP-1087](./eip-1087.md)).
```

Comment:
EIP-1087 is stagnant, which means this EIP should not reference it.  Generally speaking, you should only reference EIPs that are likely to become final *before* this EIP becomes final.

In this case, I recommend just giving a 2-line description of the alternative solution and then describe why you think this solution is superior.  No need to create a dependency on another EIP.

> Another option to solve a similar set of problems is X.  This EIP decided to do Y instead because of Z.